### DISCUSSION TITLE
Partition handling in the client

### DISCUSSION THREAD

**Discussion by 0UserName** [rabbitmq/rabbitmq-dotnet-client#1198] [Q&A] [answered]
**Situation**

There is a cluster consisting of three nodes. As a result of an emergency situation, and one of the nodes is considered partitioned, but in fact it is still running. The client accesses the cluster through load balancing, which distributes traffic randomly (port check is used as a healthcheck). `autoheal` is used as partition handling strategy. In such a scenario, it is possible that traffic will be directed to partitioned node.

**Question**

Does `rabbitmq-dotnet-client` somehow take into account the state of the cluster nodes to which it connects if specify all cluster nodes instead of VIP when connecting? Or we should rely only on the configuration of the cluster, taking into account the environment in which it is deployed, for example, replace the `autoheal` to `pause-minority`.

**Comment by michaelklishin** [comment, marked as answer]
Neither this nor other AMQP 0-9-1 clients do not treat any nodes as special. A node either allows for connections or does not (a node in the minority might stop itself, for example, or nodes [marked for maintenance](https://rabbitmq.com/upgrade.html#maintenance-mode) intentionally won't accept client connections).

This client, like most, supports lists of endpoints it will try to to connect, in order, until one of them succeeds.
